packages/data-broker/src/DataBroker.js

```diff
@@ -104,15 +106,38 @@ export class DataBroker {
     return service.pull(dataSources, type, options);
   };
 
-  mergeAnalytics = (target = { results: [], metadata: {} }, source) => ({
-    results: target.results.concat(source.results),
-    metadata: {
-      dataElementCodeToName: {
-        ...target.metadata.dataElementCodeToName,
-        ...source.metadata.dataElementCodeToName,
+  mergeAnalytics = (target = { results: [], metadata: {} }, source) => {
+    const sourceAggregationsProcessed = source.aggregationsProcessed || 0;
+    const targetResults = target.results;
+    const targetResultIndex = targetResults.findIndex(
+      ({ aggregationsProcessed }) => aggregationsProcessed === sourceAggregationsProcessed,
+    );
+    const targetResult = targetResultIndex >= 0 ? targetResults[targetResultIndex] : undefined;
+
+    const newResults = targetResult
+      ? targetResults
+          .slice(0, targetResultIndex)
+          .concat([
+            {
+              ...targetResult,
+              analytics: targetResult.analytics.concat(source.results),
+            },
+          ])
+          .concat(targetResults.slice(targetResultIndex + 1, targetResults.length - 1))
+      : targetResults.concat([
+          { analytics: source.results, aggregationsProcessed: sourceAggregationsProcessed },
+        ]);
+
+    return {
+      results: newResults,
+      metadata: {
+        dataElementCodeToName: {
+          ...target.metadata.dataElementCodeToName,
+          ...source.metadata.dataElementCodeToName,
+        },
```

Comment:
Changed the return signature a bit here! 
Previous:
```
{
  results: analytics,
  metadata: { ... },
}
```
Now:
```
{
  results: [{ analytics, aggregationsProcessed (number)}, ... ],
  metadata: { ... }
}
```
Basically, now instead of returning one set of analytics that is the merged result of all dataSources, we return an array of analytics batches that are merged by the number of aggregations they have processed. This allows aggregator to then run the remaining aggregations for each batch based on the number they've already processed.

Initially I didn't support this, and instead I just flagged to the datasources that they cannot process aggregations if the original fetch had multiple datasources in it. However, having multiple datasources is actually quite a common thing to do, especially when combining data elements and indicators in a fetch! So I felt it was worth it to support this, even if it makes the code a bit tricker to follow?

Reply:
Yep agreed. Having a comment above the function that made it obvious what was happening means it wouldn't even need to be fully followed 🙏

Reply:
I wonder if rather than the pure number of aggregations processed, data-api should be passing back the sub-array of aggregations processed, as it was passed in? Now that the response format has changed here, it could be nice to avoid baking in the same set of aggregations being run across all data sources. Just thinking of our discussion today, in case we ever wanted to push poly-aggregation support into the data services.